File: python/komputer.py
from cpu import Cpu
from ram import Ram
from harddrive import Harddrive

class Komputer:
    def __init__(self, nama: str, cpu: Cpu, ram_list: list, harddrive: Harddrive):
        self.nama = nama
        self.cpu = cpu
        self.ram_list = ram_list.copy()
        self.harddrive = harddrive

    def add_ram(self, ram: Ram):
        self.ram_list.append(ram)

    def tampilkan_info(self):
        print("\nInformasi Komputer:")
        print(f"Nama: {self.nama}")
        self.cpu.info()
        for ram in self.ram_list:
            ram.info()
        self.harddrive.info()

File: java/Komputer.java
import java.util.ArrayList;

class Komputer {
    private String nama;
    private Cpu cpu;
    private ArrayList<Ram> ramList;
    private Harddrive harddrive;

    public Komputer(String nama, Cpu cpu, ArrayList<Ram> ramList, Harddrive harddrive) {
        this.nama = nama;
        this.cpu = cpu;
        this.ramList = new ArrayList<>(ramList);
        this.harddrive = harddrive;
    }

    public void addRam(Ram ram) {
        ramList.add(ram);
    }

    public void tampilkanInfo() {
        System.out.println("\nInformasi Komputer: ");
        System.out.println("Nama: " + nama);
        cpu.info();
        for (Ram ram : ramList) {
            ram.info();
        }
        harddrive.info();
    }
}
